src/movies.js

```diff
@@ -1,16 +1,78 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(movies){
+    const mappedArr = movies.map(function(movie){
+      return movie.director
+    })
+    return mappedArr
+}
+
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies){
+    const findDirDram = movies.filter(function(movie){
+      if(movie.director === "Steven Spielberg" && movie.genre.includes("Drama")){
+        return movie
+      } 
+      
+    }) 
+      return findDirDram.length
+  }
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movies){
+    const sumOfrates = movies.reduce(function(total, val){
+      if(val.rate === ''|| val.rate === undefined){
```

Comment:
Don't think you need the or in here. Could just use `if(!val.rate)` which just means if the value gives you a falsy value.